modules/autocomplete-picker/main/index.coffee

```diff
@@ -0,0 +1,210 @@
+
+class AutocompletePicker extends hx.EventEmitter
+  constructor: (selector, items, options) ->
+    super()
+
+    hx.component.register(selector, this)
+
+    defaults =
+      # Options passed to the feed - defaults defined there
+      filter: undefined
+      filterOptions: undefined
+      matchType: undefined
+      showOtherResults: undefined
+      trimTrailingSpaces: undefined
+      valueLookup: undefined
+
+      # Options used by the picker
+      buttonClass: undefined
+      disabled: false
+      renderer: undefined
+      value: undefined
+      chooseValueText: hx.userFacingText('autocompletePicker', 'chooseValue')
+      loadingText: hx.userFacingText('autocompletePicker', 'loading')
+      noResultsText: hx.userFacingText('autocompletePicker', 'noResults')
+      otherResultsText: hx.userFacingText('autocompletePicker', 'otherResults')
+
+    resolvedOptions = hx.merge defaults, options
+
+    selection = hx.select(selector)
+      .classed('hx-autocomplete-picker hx-btn', true)
+
+    if resolvedOptions.buttonClass
+      selection.classed(resolvedOptions.buttonClass, true)
+
+    valueText = selection.append('div').class('hx-autocomplete-picker-text')
+    selection.append('span').class('hx-autocomplete-picker-icon')
+      .append('i').class('hx-icon hx-icon-caret-down')
+
+    feedOptions =
+      filter: resolvedOptions.filter
+      filterOptions: resolvedOptions.filterOptions
+      matchType: resolvedOptions.matchType
+      showOtherResults: resolvedOptions.showOtherResults
+      trimTrailingSpaces: resolvedOptions.trimTrailingSpaces
+      valueLookup: resolvedOptions.valueLookup
+
+    feed = new hx.AutocompleteFeed(feedOptions)
+
+    @_ =
+      options: resolvedOptions
+      valueText: valueText
+      feed: feed
+      valueLookup: resolvedOptions.valueLookup or hx.identity
+
+    if validateItems(feed, items)
+      feed.items(items)
+
+      renderWrapper = (element, item) =>
+        selection = hx.select(element)
+          .clear()
+          .classed('hx-autocomplete-picker-heading', item.heading)
+        if item.unselectable or item.heading
+          hx.select(element)
+            .text(item.text)
+            .off()
+        else
+          @_.renderer(element, item)
+
+      menu = new hx.Menu(selector, {
+        dropdownOptions:
+          ddClass: 'hx-autocomplete-picker-dropdown'
+      })
+
+      @_.renderer = resolvedOptions.renderer or menu.renderer()
+      @_.menu = menu
+      menu.renderer(renderWrapper)
+
+      noResultsItem =
+        text: resolvedOptions.noResultsText
+        unselectable: true
+
+      loadingItem =
+        text: resolvedOptions.loadingText
+        unselectable: true
+
+      otherResultsItem =
+        text: resolvedOptions.otherResultsText
+        unselectable: true
+        heading: true
+
+      renderMenu = (items) ->
+        menu.items(items)
+        menu.dropdown._.setupDropdown(menu.dropdown._.dropdown.node())
```

Comment:
This function will not be required once #180 has been completed as `menu.items(items)` should re-render the dropdown if it is called when the dropdown is open.